include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -726,14 +726,25 @@ struct __parallel_reduce_then_scan_scan_submitter
     _InitType __init;
 };
 
-// reduce_then_scan requires subgroup size of 32, and performs well only on devices with fast coordinated subgroup
-// operations.  We do not want to run this scan on CPU targets, as they are not performant with this algorithm.
+// With optimization enabled, reduce-then-scan requires a sub-group size of 32. Without optimization, we must compile
+// to a sub-group size of 16 to workaround a hardware bug on certain Intel integrated graphics architectures.
+constexpr inline std::uint8_t __reduce_then_scan_sg_sz =
+#if _ONEDPL_DETECT_COMPILER_OPTIMIZATIONS_ENABLED
+    std::uint8_t{32};
+#else
+    std::uint8_t{16};
+#endif
```

Comment:
Perhaps this is just my preference, but I think I would prefer this as a constexpr function we could call, rather than inline `std::uint8_t` just floating in the namespace.

Reply:
Sure, I made this change.